### PR TITLE
Add support for IBM cloud platform load balancers

### PR DESCRIPTION
**Pull request by AndreaM12345**
Need to get approval for https://github.com/openshift/api/pull/536 and then can update the glide to pull in the new Platform

### CODE REVIEW

File: pkg/operator/controller/ingress/load_balancer_service.go [hunk 1]
```diff
@@ -53,6 +57,9 @@ var (
 		configv1.NonePlatformType: nil,
 		// vSphere does not support load balancers as of 2019-06-17.
 		configv1.VSpherePlatformType: nil,
+		configv1.IBMCloudPlatformType: {
+			"service.kubernetes.io/ibm-load-balancer-cloud-provider-ip-type": "private",
```
Comment (ironcladlou) [inline pkg/operator/controller/ingress/load_balancer_service.go]: Looks like this should use the new `iksLBProxyProtocolAnnotations` constant?

File: pkg/operator/controller/ingress/load_balancer_service.go [hunk 2]
```diff
@@ -110,10 +117,17 @@ func desiredLoadBalancerService(ci *operatorv1.IngressController, deploymentRef
 	if infraConfig.Status.Platform == configv1.AWSPlatformType {
 		service.Annotations[awsLBProxyProtocolAnnotation] = "*"
 	}
+
+	if infraConfig.Status.Platform == configv1.IBMCloudPlatformType {
+		service.Annotations[iksLBProxyProtocolAnnotations] = "public"
+		//service.Annotations[iksLBProxyRegionAnnotations] = infraConfig.Status.PlatformStatus.IBMCloud.Location
```
Comment (ironcladlou) [inline pkg/operator/controller/ingress/load_balancer_service.go]: Leftover commented line can be removed

File: pkg/operator/controller/ingress/load_balancer_service.go [hunk 3]
```diff
@@ -110,10 +117,17 @@ func desiredLoadBalancerService(ci *operatorv1.IngressController, deploymentRef
...
+	}
+
 	if isInternal {
 		annotation := InternalLBAnnotations[infraConfig.Status.Platform]
 		for name, value := range annotation {
 			service.Annotations[name] = value
+
```
Comment (ironcladlou) [inline pkg/operator/controller/ingress/load_balancer_service.go]: Extra newline

**Comment by ironcladlou** [comment]
/retitle Add support for IBM cloud platform load balancers

**Comment by ironcladlou** [comment]
Thanks for this!

/lgtm

**Comment by ironcladlou** [comment]
@csrwng we can't cherry pick this without back porting the API...

**Comment by AndreaM12345** [comment]
trying to back port the API with the following https://github.com/openshift/api/pull/580

**Comment by derekwaynecarr** [comment]
@ironcladlou @csrwng @AndreaM12345 just tagged the backport for 4.3.2 of API, we can create a cherry pick now.

**Comment by AndreaM12345** [comment]
created https://github.com/openshift/cluster-ingress-operator/pull/355 to cherry pick the changes into 4.3

**Pull request merged**